docs/src/userguide/extension_types.rst

```diff
@@ -76,6 +76,51 @@ and the depth attribute readable but not writable.
     Python access, not direct access. All the attributes of an extension type
     are always readable and writable by C-level access.
 
+
+.. _dynamic_attributes:
+
+Dynamic Attributes
+==================
+
+It is not possible to add attributes to an extension type at runtime by default.
+You have two ways of avoiding this limitation, both add an overhead when
+a method is called from Python code.
+
+The first workaround is making a child Python class and is preferred way of
```

Comment:
'workaround' sounds negative, like this is something that users need to fix because Cython fails to do it for them, whereas it's actually an efficiency boost for an object not to have a dict (it reduces the memory consumption quite a bit, for example).

Better: "The first approach is to create a Python subclass."

I'd also not name it the "preferred way". There might be reasons to prefer one over the other, but my guess is that it's rather the `__dict__` than the subclassing approach that is more efficient. It avoids creating a second (Heap-)type, for one. Haven't done any measurements, though, so I can't really judge.

Reply:
I've actually done measurements, and for cpdef, subclassing is faster (15%) than the dict approach. This is why I've put `the preferred way`. But sure, I'll change it.

Reply:
Interesting. Then, don't name it "preferred" (as I'd like to avoid giving a recommendation purely based on speed), but just mention that both have a penalty, and that that of a subclass can be smaller.

Reply:
To clarify, I'd like to avoid giving a recommendation specifically here, because these two approaches are really so different. The performance implications are just a small part of the implications to consider.